pkg/streamingpromql/optimize/ast/util_test.go

```diff
@@ -28,19 +28,29 @@ func preprocessQuery(t *testing.T, expr parser.Expr) (parser.Expr, error) {
 	return promql.PreprocessExpr(expr, timestamp.Time(dummyTimeRange.StartT), timestamp.Time(dummyTimeRange.EndT), 0)
 }
 
-func getOutputFromASTOptimizationPassWithQueryPlan(t *testing.T, ctx context.Context, input string, createOptimizerFunc func(prometheus.Registerer) optimize.ASTOptimizationPass) (*prometheus.Registry, parser.Expr) {
-	dummyTimeRange := types.NewInstantQueryTimeRange(timestamp.Time(1000))
-	reg := prometheus.NewPedanticRegistry()
+func runASTOptimizationPass(t *testing.T, ctx context.Context, input string, createOptimizerFunc func(prometheus.Registerer) optimize.ASTOptimizationPass) (*prometheus.Registry, parser.Expr) {
```

Comment:
I think we can simplify this further: `runASTOptimizationPassWithoutMetrics` can call `runASTOptimizationPass` passing a `createOptimizerFunc` value that just returns the optimization pass provided to `runASTOptimizationPassWithoutMetrics`. Then there's no need for `runASTOptimizationPassInner`.